autoware.proj.repos

```diff
@@ -2,53 +2,57 @@ repositories:
   # autoware
   autoware/pilot.auto:
     type: git
-    url: git@github.com:tier4/pilot.auto.git
+    url: https://github.com/tier4/pilot.auto.git
     version: ros2
   autoware/launcher:
     type: git
-    url: git@github.com:tier4/autoware_launcher.iv.universe.git
+    url: https://github.com/tier4/autoware_launcher.iv.universe.git
     version: ros2
   # description
   description/sensor/sensor_description:
     type: git
-    url: git@github.com:tier4/sensor_description.iv.universe.git
+    url: https://github.com/tier4/sensor_description.iv.universe.git
     version: ros2
   description/vehicle/lexus_description:
     type: git
-    url: git@github.com:tier4/lexus_description.iv.universe.git
+    url: https://github.com/tier4/lexus_description.iv.universe.git
     version: ros2
   # simulator
   simulator/planning_simulator:
     type: git
-    url: git@github.com:tier4/planning_simulator.iv.universe.git
+    url: https://github.com/tier4/planning_simulator.iv.universe.git
     version: ros2
   simulator/planning_simulator_launcher:
     type: git
-    url: git@github.com:tier4/planning_sim_launcher.iv.universe.git
+    url: https://github.com/tier4/planning_sim_launcher.iv.universe.git
     version: ros2
   simulator/scenario_runner:
     type: git
-    url: git@github.com:tier4/scenario_runner.iv.universe.git
+    url: https://github.com/tier4/scenario_runner.iv.universe.git
     version: ros2
+  vendor/osqp:
+    type: git
+    url: https://github.com/tier4/osqp_vendor.git
```

Comment:
Ideally we should make sure that the user of osqp_interface (e.g. mpc_follower) can also succeeds to build & run with these libraries, but I think we can create a bug-fix pull request once we notice any issues.

I've checked that osqp_interface from this PR https://github.com/tier4/Pilot.Auto/pull/14 succeeds to build with this osqp_vendor. Since this shouldn't break any build at this point, I will merge this PR so that we can try it with osqp_interface and mpc_follower.

Reply:
@mitsudome-r @esteve I started testing last night and continued this morning. It took me too long to test exacly what you asked for. Now you merged in the meantime. In my experience, it doesn't work. I suspect you might have checked with `osqp` installed in `/usr/local/lib`. I created an issue that needs to be adressed https://github.com/tier4/osqp_vendor/issues/2

Reply:
@fred-apex-ai CI was passing for your PR after adding `osqp_vendor` to the the CI workflow (https://github.com/tier4/Pilot.Auto/pull/14/checks?check_run_id=1260029946), the CI runner does not have osqp installed system-wide, it now uses `osqp_vendor` and exports the libraries to the downstream packages.

Reply:
> You turn on building shared libs explicitly in https://github.com/tier4/osqp_vendor/blob/main/CMakeLists.txt#L15 Do we still have the static objects as well?

Since `osqp_vendor` reexports the `osqp` target from `osqp`, both `osqp::osqp` and `osqp::osqpstatic` are available to downstream packages.

This line https://github.com/tier4/osqp_vendor/blob/main/CMakeLists.txt#L15 can be removed, since `osqp` by default builds both shared and static libraries.